app/models/metric/capture.rb

```diff
@@ -128,32 +127,40 @@ def self.perf_capture_health_check(zone)
   end
   private_class_method :perf_capture_health_check
 
+  # Collect realtime targets and group them by their rollup parent
+  #
+  # 1. Only calculate rollups for Hosts
+  # 2. Some Hosts have an EmsCluster as a parent, others have none.
+  # 3. Only Hosts with a parent are rolled up.
+  #
+  # @returns Hash<String,Array<Host>>
+  #   e.g.: {"EmsCluster:4"=>[Host:4], "EmsCluster:5"=>[Host:1, Host:2]}
   def self.calc_targets_by_rollup_parent(targets)
-    # Collect realtime targets and group them by their rollup parent, e.g. {"EmsCluster:4"=>[Host:4], "EmsCluster:5"=>[Host:1, Host:2]}
-    targets_by_rollup_parent = targets.inject({}) do |h, target|
-      next(h) unless target.kind_of?(Host) && perf_capture_now?(target)
-
-      interval_name = perf_target_to_interval_name(target)
-      next unless interval_name == "realtime"
-
-      target.perf_rollup_parents(interval_name).to_a.compact.each do |parent|
+    realtime_targets = targets.select do |target|
+      target.kind_of?(Host) &&
+      perf_target_to_interval_name(target) == "realtime" &&
+      perf_capture_now?(target)
```

Comment:
Minor but can you indent these last two lines to show they are part of the entire conditional?

```ruby
  realtime_targets = targets.select do |target|
    target.kind_of?(Host) &&
      perf_target_to_interval_name(target) == "realtime" &&
      perf_capture_now?(target)
  end
```